servant/src/Servant/Types/SourceT.hs

```diff
@@ -25,7 +25,7 @@ import           System.IO
 import qualified Test.QuickCheck            as QC
 
 -- $setup
--- >>> :set -XOverloadedStrings
+-- >>> import Data.String (fromString)
```

Comment:
Doctest v0.21 (the first series compatible with GHC 9.6) has a better treatment of warnings. So the type-defaulting was causing this warning to fail the test:
```
src/Servant/Types/SourceT.hs:223: failure in expression `runSourceT (source "foo" :: SourceT Identity Char)'
expected: ExceptT (Identity (Right "foo"))
 but got: 
          ^
          <interactive>:263:13: warning: [GHC-18042] [-Wtype-defaults]
              • Defaulting the type variable ‘f0’ to type ‘[]’ in the following constraint
                  Foldable f0 arising from a use of ‘source’
              • In the first argument of ‘runSourceT’, namely
                  ‘(source "foo" :: SourceT Identity Char)’
                In the expression:
                  runSourceT (source "foo" :: SourceT Identity Char)
                In an equation for ‘it’:
                    it = runSourceT (source "foo" :: SourceT Identity Char)
```

This is probably happening since #1593.

One option would be to turn of warnings for type-defaulting, but I really like the explicit types, so I opted for that instead. I am not sure what other instances come into play here, but I think its nice that the user doesn't have to think about that. Type defaulting might be fine for GHCi but I think it should be avoided for doctests.